providers/hns/hns_roce_u_hw_v2.c

```diff
@@ -766,6 +766,122 @@ static __le32 get_immtdata(enum ibv_wr_opcode opcode, const struct ibv_send_wr *
 	}
 }
 
+static int check_ud_opcode(struct hns_roce_ud_sq_wqe *ud_sq_wqe,
+			   const struct ibv_send_wr *wr)
+{
+	uint32_t ib_op = wr->opcode;
+
+	if (ib_op != IBV_WR_SEND && ib_op != IBV_WR_SEND_WITH_IMM)
+		return EINVAL;
```

Comment:
Error returned from the functions should be < 0. I see that you are returning > 0 already, but it doesn't look right.

Reply:
I use positive return values because I found the following disscusion before: https://www.spinics.net/lists/linux-rdma/msg89507.html, and according to the man page, ibv_post_send() should return a positive errno. Did I misunderstand it?
Thank you.

Reply:
Jason talks about "errno = ...", which is returned to the user, while here we are talking about internal return value from internal functions. 

For example, this is why librdmacm has ERR() macro.
https://github.com/linux-rdma/rdma-core/blob/master/librdmacm/cma.h#L89
It separates "errno" and "ret".

The current hns code is correct, but misleading.

Reply:
Thanks for your explanation, we will fix this later.